File: memento/python/text_undo_care_taker.py
class TextUndoCareTaker:
    """
    Memento'ların referansının tutulduğu yerdir.
    UML diyagramındaki CareTaker yapısına denk gelmektedir.
    """

    def __init__(self):
        self._mementos = []

    # Çağrılma işlemi yapıldığında yığının en üstündeki Memento örneği silinir ve geriye döndürülür.

    # Ekleme işlemi yapıldığında yığının en üstüne Memento örneği eklenir.
    # Klasik Stack.
    @property
    def text_memento(self):
        return self._mementos.pop()

    @text_memento.setter
    def text_memento(self, value):
        self._mementos.append(value)

File: memento/typescript/text-undo-care-taker.ts
import {TextMemento} from "./text-memento";

// Memento'ların referansının tutulduğu yerdir.
// UML diyagramındaki CareTaker yapısına denk gelmektedir.
export class  TextUndoCareTaker {
  private mementos: TextMemento[] = [];

  // Çağrılma işlemi yapıldığında yığının en üstündeki Memento örneği silinir ve geriye döndürülür.
  get textMemento() {
    return this.mementos.pop();
  }

  // Ekleme işlemi yapıldığında yığının en üstüne Memento örneği eklenir.
  set textMemento(value) {
    this.mementos.push(value);
  }
}
